csrc/scheduler/vectorize_helper.cpp

```diff
@@ -365,9 +392,48 @@ std::vector<IterDomain*> ContiguousInnerDimensionsMapper::projectId(
     distributePE(merge_or_split);
   };
 
-  auto clear_left_of = [&frontier](IterDomain* id) {
-    auto it = std::find(frontier.begin(), frontier.end(), id);
-    if (it != frontier.end()) {
+  auto propagateResize = [&frontier, this](Resize* resize_op, bool p2c) {
+    // TODO: resize could affect the offset/pad_extent impacting vectorization
+    // factor on the loading inputs. But we shouldn't use that to block the
+    // propagation. We'll resolve it in getTvToContigMergeOfInnerSizeMap when we
+    // compute the actual vectorization factor on the TensorView affected.
+    IterDomain* id_from = p2c ? resize_op->in() : resize_op->out();
+    IterDomain* id_to = p2c ? resize_op->out() : resize_op->in();
+
+    auto it = std::find(frontier.begin(), frontier.end(), id_from);
+    if (it == frontier.end()) {
+      return;
+    }
+
+    auto pos = std::distance(frontier.begin(), it);
+    if (resize_in_pad_.count(resize_op) != 0) {
+      // resize created by PadOp.
+
+      // project resize op to frontier.
+      frontier[pos] = id_to;
+      // clear left of resize, since those are no long contiguous.
+      frontier.erase(frontier.begin(), it);
+
+      if (recording_) {
+        // since this is a pad operation, we don't need to worry about the
+        // offset, but rather the pad_extent, which becomes the real buffer on
```

Comment:
What do you refer to by "offset"?

Reply:
I was referring to the offset created by slice. i.e. negative resize extent.

Reply:
"Negative resize extent" sounds less ambiguous.